apps/src/codebridge/Console/ConsoleManager.ts

```diff
@@ -0,0 +1,78 @@
+import {FitAddon} from '@xterm/addon-fit';
+import {Terminal} from '@xterm/xterm';
+
+// Manager for xterm.js-based console in codebridge
+export default class ConsoleManager {
+  private terminal: Terminal;
+  private terminalFitAddon: FitAddon;
+  private terminalLines: string[];
+
+  constructor(terminal: Terminal, terminalFitAddon: FitAddon) {
+    this.terminal = terminal;
+    this.terminalFitAddon = terminalFitAddon;
+    this.terminalLines = [];
+  }
+
+  public getTerminal() {
+    return this.terminal;
+  }
+
+  public getTerminalFitAddon() {
+    return this.terminalFitAddon;
+  }
+
+  public setTerminal(terminal: Terminal) {
+    this.terminal = terminal;
+  }
+
+  public setTerminalFitAddon(terminalFitAddon: FitAddon) {
+    this.terminalFitAddon = terminalFitAddon;
+  }
+
+  public clearTerminalLines() {
+    this.terminalLines = [];
+    this.terminal.clear();
+  }
+
+  public getTerminalLines() {
+    return this.terminalLines;
+  }
+
+  public writeConsoleMessage(message: string) {
+    const lines = message.split('\n');
+    lines.forEach(l => this.appendTerminalLine(l));
+  }
+
+  public writeSystemMessage(message: string, appName: string) {
+    this.writeConsoleMessage(this.getSystemMessage(message, appName));
+  }
+
+  public writeErrorMessage(message: string) {
+    // This colors the message red in the terminal
+    this.writeConsoleMessage(`\x1b[31m${message}\x1b[0m`);
+  }
+
+  public writeSystemError(message: string, appName: string) {
+    this.writeErrorMessage(this.getSystemMessage(message, appName));
+  }
+
+  public writeImage(base64Image: string) {
+    const dataSize = atob(base64Image).length;
+    // This is a special sequence that tells the terminal to display an image
+    // See documentation here: https://iterm2.com/documentation-images.html
+    const imageString = `\x1b]1337;File=inline=1;size=${dataSize};width=400px;height=400px:${base64Image}\x1b\\`;
```

Comment:
Could use constant for image dimensions as we may tweak them.

Reply:
done!